### PR TITLE
Enhancement pool report tab changes

### PR DESCRIPTION
**Pull request by samu80**
Trello link: https://trello.com/c/n9GFNU3B

### CODE REVIEW

File: app/views/pool_reports/_list.html.haml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            %td= emp_detail.try(:remark)
+            %td= emp_detail.try(:oeta)
+            %td= emp_detail.try(:neta)
+            %td= user.get_owner_names(emp_detail.deployment_owner_emails)
+            %td= user.get_owner_names(emp_detail.owned_by_emails)
+            %td= emp_detail.try(:cv_link)
+            %td= "PIP"
+            %td= "ROI"
+            %td= "Mentor for Training"
+            %td= user.get_project_names(user_projects.where(:active => false)).join(", ")
```
Comment (RohitNalePUCSD) [inline app/views/pool_reports/_list.html.haml:56]: already we are fetched inactive projects.

File: app/views/pool_reports/_list.html.haml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            %td= emp_detail.try(:interview_rejected)
+            - if edit_access
+              %td= link_to '', edit_pool_report_path(user.id), data: {'no-turbolink' =>  true}, "data-toggle" => "modal", 'data-target' => '#showEditModal', title: "Edit Project", class: "icon-edit", locals: { user: user }, remote: true
+
+:javascript
+  $('.usersList').popover();
+  $('#sortable').dataTable({ 'aaSorting': [], 'scrollX': true });
```
Comment (RohitNalePUCSD) [inline app/views/pool_reports/_list.html.haml:63]: change the datatable properties
$('#sortable').dataTable({ 'aaSorting': [], 'scrollX': true,  'iDisplayLength': 50, 'lengthMenu': [50, 100, 150, 200] });

File: app/views/pool_reports/edit.js.erb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+$('#showEditModal').html("<%= j render(:partial => 'edit_form', :locals => { :user => @user }) %>");
+//$('#editModal').modal('show')
```
Comment (RohitNalePUCSD) [inline app/views/pool_reports/edit.js.erb:2]: remove commented code

File: app/views/report_mailer/export_bench_resources_report.xlsx.axlsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
+wb = xlsx_package.workbook
+wb.add_worksheet(name: "PoolReport") do |sheet|
+  sheet.add_row %w(CurrentProject RAG EmployeeID Name CoreSkills OtherSkills
```
Comment (RohitNalePUCSD) [inline app/views/report_mailer/export_bench_resources_report.xlsx.axlsx:3]: Change the coulmns seq, 1st column is id and name

File: app/views/users/_deployments.html.haml [hunk 5]
```diff
@@ -0,0 +1,19 @@
+- disabled = (EmployeeDetail::DEPLOYMENT_INFO_EDIT_USER_ROLES.include?(current_user.role) || current_user.email.eql?(DEPLOYMENT_TEST_USER)) ? false : true
+= simple_form_for(@user,:url => user_path(@user), :html => { method: :put, class: "form-horizontal", multipart: true}) do |f|
+  = f.simple_fields_for :employee_detail do |e|
+    = e.input :available_from, label: "Available From", input_html: { type: 'date', class: 'date-picker', value: @user.employee_detail.available_from.try(:strftime, "%Y-%m-%d"), disabled: disabled}
+    = e.input :cv_link, label: 'CV Link', value: @user.employee_detail.cv_link, input_html: { disabled: disabled }
+    = e.input :deployment_owner_emails, label: 'Deployment Owner Emails', collection: @deployment_owner_emails, input_html: { class: 'deployment_owner_emails', disabled: disabled, multiple: true, style: "width:300px;", "data-placeholder" => "You can add multiple emails" }
```
Comment (RohitNalePUCSD) [inline app/views/users/_deployments.html.haml:6]: changed single select not multselect

File: config/initializers/global_constants.rb [hunk 6]
```diff
@@ -174,3 +174,5 @@
 
 PROJECT_TEAM_REPORT_TYPE = ['Forecast', 'MonthEnd']
 UI_UX_AND_DEVOPS_PROJECTS = ['UI-UX Work', 'DevOps Work']
+DEPLOYMENT_OWNER_EMAILS = ['rohit.nale@joshsoftware.com', 'swapnil@joshsoftware.com']
```
Comment (RohitNalePUCSD) [inline config/initializers/global_constants.rb:177]: Remove the test user from constant and add the corrected user.

File: config/routes.rb [hunk 7]
```diff
@@ -87,6 +87,11 @@
     end
     get 'export_project_report', as: :export_report, on: :collection
   end
+  resources :pool_reports, only: [:index, :edit, :update] do
+    get 'export_pool_reports', as: :export_pool_reports, on: :collection
+    #get 'edit_modal', as: :edit_modal, on: :collection
```
Comment (RohitNalePUCSD) [inline config/routes.rb:92]: remove commented code

File: lib/tasks/calculate_peroject_allocation_daily.rake [hunk 8]
```diff
@@ -0,0 +1,20 @@
+desc 'Calculate project allocation on every day'
+task :calculate_project_allocation => :environment do
```
Comment (RohitNalePUCSD) [inline lib/tasks/calculate_peroject_allocation_daily.rake:2]: Add this rake task in the schedular file at 6:30 every day.

File: lib/tasks/calculate_peroject_allocation_daily.rake [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+  users = User.approved_and_on_notice.where(:'employee_detail.division' => DIVISION_TYPES[:development])
+
+  users.each do |user|
+    active_allocation_project = user.user_projects.where(active: true, :end_date.gt => Date.today)
+    total_allocation = active_allocation_project.sum(:allocation)
+
+    if total_allocation < 160 && user.employee_detail.present?
+      user.employee_detail.available_from = Date.today   if user.employee_detail.available_from.nil? # need to check which date is entry.
```
Comment (RohitNalePUCSD) [inline lib/tasks/calculate_peroject_allocation_daily.rake:10]: Change code:
`min_end_date_project = user.user_projects.where(active: false).sort_by(&:end_date).last
if min_end_date_project.present?
    user.employee_detail.available_from = min_end_date_project.end_date + 1.day
 else
   user.employee_detail.available_from = user.private_profile.date_of_joining
end `

File: lib/tasks/hono_integration.rake [hunk 10]
```diff
@@ -206,8 +206,9 @@ namespace :hono_integration do
       "division": @employee_params["department"],
       "sub_business_unit": @employee_params["Sub_Business_Unit"]
     }
-    if @employee.new_record? && employee_detail.new_record? && employee_detail.eligible_for_assessment? && @employee_params["date_of_joining"].present?
-      employee_detail_attribute["assessment_month"] = calculate_assessment_months
+    if @employee.new_record? && employee_detail.new_record?
+      employee_detail.available_from = Date.today
+      employee_detail_attribute["assessment_month"] = calculate_assessment_months if employee_detail.eligible_for_assessment? && @employee_params["date_of_joining"].present?
```
Comment (RohitNalePUCSD) [inline lib/tasks/hono_integration.rake:211]: add available_hours = 160

**Review by RohitNalePUCSD — APPROVED** [review]

File: lib/tasks/calculate_peroject_allocation_daily.rake (hunk 8 above)
Comment (samu80) [reply lib/tasks/calculate_peroject_allocation_daily.rake:2]: Added.

File: lib/tasks/hono_integration.rake (hunk 10 above)
Comment (samu80) [reply lib/tasks/hono_integration.rake:212]: Added

**Review by RohitNalePUCSD — APPROVED** [review]

**Pull request merged**